app/src/main/java/com/duckduckgo/app/di/DatabaseModule.kt

```diff
@@ -21,31 +21,20 @@ import android.webkit.WebViewDatabase
 import androidx.room.Room
 import androidx.room.RoomDatabase
 import com.duckduckgo.app.browser.addtohome.AddToHomeCapabilityDetector
-import com.duckduckgo.app.browser.httpauth.RealWebViewHttpAuthStore
-import com.duckduckgo.app.browser.httpauth.WebViewHttpAuthStore
-import com.duckduckgo.app.fire.DatabaseCleaner
-import com.duckduckgo.app.fire.DatabaseLocator
-import com.duckduckgo.app.global.DispatcherProvider
 import com.duckduckgo.app.global.db.AppDatabase
 import com.duckduckgo.app.global.db.MigrationsProvider
 import com.duckduckgo.app.settings.db.SettingsDataStore
 import dagger.Module
 import dagger.Provides
-import javax.inject.Named
 import javax.inject.Singleton
 
 @Module(includes = [DaoModule::class])
 class DatabaseModule {
 
     @Provides
     @Singleton
-    fun provideWebViewHttpAuthStore(
-        context: Context,
-        databaseCleaner: DatabaseCleaner,
-        @Named("authDbLocator") authDatabaseLocator: DatabaseLocator,
-        dispatcherProvider: DispatcherProvider
-    ): WebViewHttpAuthStore {
-        return RealWebViewHttpAuthStore(WebViewDatabase.getInstance(context), databaseCleaner, authDatabaseLocator, dispatcherProvider)
```

Comment:
I changed the approach to provide stubs for the `DispatcherProvider`, as it was way simpler to do it that way. You'll see I added a `StubCoroutineModule` that does that.
In that way, the `RealWebViewHttpAuthStore` is always taken from the real dagger module, and when in test, it takes the stub dispatcher provider